pkg/controller/webhooks/default_handler.go

```diff
@@ -0,0 +1,263 @@
+		if quotaErrs != nil {
+			allErrs = append(allErrs, &field.Error{
+				Field: "quota",
+				Detail: fmt.Sprintf("quota check failed: %s",
+					quotaErrs.Error()),
+				BadValue: "*your namespace quota*",
+				Type:     field.ErrorTypeForbidden,
+			})
+		}
+	}
+	err := r.validateResourceNameLength(comp.GetName(), r.length)
+	if err != nil {
+		allErrs = append(allErrs, &field.Error{
+			Field: ".metadata.name",
+			Detail: fmt.Sprintf("Please shorten Redis name, currently it is: %s",
+				err.Error()),
+			BadValue: comp.GetName(),
+			Type:     field.ErrorTypeTooLong,
+		})
+	}
+
+	// We aggregate and return all errors at the same time.
+	// So the user is aware of all broken parameters.
+	// But at the same time, if any of these fail we cannot do proper quota checks anymore.
+	if len(allErrs) != 0 {
+		return nil, apierrors.NewInvalid(
+			r.gk,
+			comp.GetName(),
+			allErrs,
+		)
+	}
+
+	return nil, nil
+}
+
+// ValidateUpdate implements webhook.CustomValidator so a webhook will be registered for the type
+func (r *DefaultWebhookHandler) ValidateUpdate(ctx context.Context, oldObj, newObj runtime.Object) (admission.Warnings, error) {
+	allErrs := field.ErrorList{}
+	comp, ok := newObj.(common.Composite)
+	if !ok {
+		return nil, fmt.Errorf("provided manifest is not a valid " + r.gk.Kind + " object")
+	}
+
+	if comp.GetDeletionTimestamp() != nil {
+		return nil, nil
+	}
+
+	if r.withQuota {
+		quotaErrs := r.checkQuotas(ctx, comp, true)
+		if quotaErrs != nil {
+			allErrs = append(allErrs, &field.Error{
+				Field: "quota",
+				Detail: fmt.Sprintf("quota check failed: %s",
+					quotaErrs.Error()),
+				BadValue: "*your namespace quota*",
+				Type:     field.ErrorTypeForbidden,
+			})
+		}
+	}
+
+	err := r.validateResourceNameLength(comp.GetName(), r.length)
+	if err != nil {
+		allErrs = append(allErrs, &field.Error{
+			Field: ".metadata.name",
+			Detail: fmt.Sprintf("Please shorten "+cases.Title(language.English).String(r.name)+" name, currently it is: %s",
+				err.Error()),
+			BadValue: comp.GetName(),
+			Type:     field.ErrorTypeTooLong,
+		})
+	}
+
+	// We aggregate and return all errors at the same time.
+	// So the user is aware of all broken parameters.
+	// But at the same time, if any of these fail we cannot do proper quota checks anymore.
+	if len(allErrs) != 0 {
+		return nil, apierrors.NewInvalid(
+			r.gk,
+			comp.GetName(),
+			allErrs,
+		)
+	}
+
+	return nil, nil
+}
+
+// ValidateDelete implements webhook.CustomValidator so a webhook will be registered for the type
+func (r *DefaultWebhookHandler) ValidateDelete(ctx context.Context, obj runtime.Object) (admission.Warnings, error) {
+	allErrs := field.ErrorList{}
+
+	comp, ok := obj.(common.Composite)
+	if !ok {
+		return nil, fmt.Errorf("provided manifest is not a valid " + r.gk.Kind + " object")
+	}
+
+	allErrs = GetClaimDeletionProtection(comp.GetSecurity(), allErrs)
+
+	if len(allErrs) != 0 {
+		return nil, apierrors.NewInvalid(
+			nextcloudGK,
```

Comment:
I suppose this is wrong, I guess this code should be generic for any service.

Reply:
Fixed